Flaky test: test.ajax.js-http #5061 ajax returns ETIMEDOUT error on timeout:

seen it on travis and locally, tested with Node v6 and v7

```
  1) test.ajax.js-http #5061 ajax returns ETIMEDOUT error on timeout:

     Uncaught AssertionError: expected 'ESOCKETTIMEDOUT' to match /(ETIMEDOUT|ENETUNREACH|EAGAIN|ECONNREFUSED)/
      at tests/integration/test.ajax.js:19:25
      at Request._callback (node_modules/pouchdb-ajax/lib/index.js:79:14)
      at self.callback (node_modules/request/request.js:187:22)
      at ClientRequest.<anonymous> (node_modules/request/request.js:779:16)
      at Socket.socket.once (_http_client.js:566:38)
      at Socket._onTimeout (net.js:339:8)
```

--- a/tests/integration/test.ajax.js
+++ b/tests/integration/test.ajax.js
@@ -16,11 +16,10 @@ adapters.forEach(function (adapter) {
       }, function (err, res) {
         // here's the test, we should get an 'err' response
         should.exist(err);
-        err.code.should.match(/(ETIMEDOUT|ENETUNREACH|EAGAIN|ECONNREFUSED)/);
+        err.code.should.match(/(ESOCKETTIMEDOUT|ETIMEDOUT|ENETUNREACH|EAGAIN|ECONNREFUSED)/);
         should.not.exist(res);
         done();
       });
     });
   });
 });
-

--- a/tests/integration/test.basics.js
+++ b/tests/integration/test.basics.js
@@ -1109,7 +1109,7 @@ adapters.forEach(function (adapter) {
       it('#5471 PouchDB.plugin() should throw error if passed wrong type or empty object', function () {
         (function () {
           PouchDB.plugin('pouchdb-adapter-memory');
-        }).should.throw(Error, 'Invalid plugin: object passed in is empty or not an object');
+        }).should.throw(Error, 'Invalid plugin: got "pouchdb-adapter-memory", expected an object or a function');
       });
     }
   });

